deeppavlov/core/commands/train.py

```diff
@@ -205,437 +97,44 @@ def train_evaluate_model_from_config(config: [str, Path, dict], iterator=None, *
-    metrics_functions = _parse_metrics(train_config['metrics'], in_y, config['chainer']['out'])
-
-    if to_train:
-        model = fit_chainer(config, iterator)
-
-        if callable(getattr(model, 'train_on_batch', None)):
-            _train_batches(model, iterator, train_config, metrics_functions, start_epoch_num=start_epoch_num)
+    if start_epoch_num is not None:
+        train_config['start_epoch_num'] = start_epoch_num
 
-        model.destroy()
-
-    res = {}
+    if 'evaluation_targets' not in train_config and ('validate_best' in train_config
+                                                     or 'test_best' in train_config):
+        log.warning('"validate_best" and "test_best" parameters are deprecated.'
+                    ' Please, use "evaluation_targets" list instead')
 
-    if iterator is not None and (train_config['validate_best'] or train_config['test_best']):
-        model = build_model(config, load_trained=to_train)
-        log.info('Testing the best saved model')
+        train_config['evaluation_targets'] = []
+        if train_config.pop('validate_best', True):
+            train_config['evaluation_targets'].append('valid')
+        if train_config.pop('test_best', True):
+            train_config['evaluation_targets'].append('test')
 
-        if train_config['validate_best']:
-            report = {
-                'valid': _test_model(model, metrics_functions, iterator,
-                                     train_config.get('batch_size', -1), 'valid',
-                                     show_examples=train_config['show_examples'])
-            }
+    trainer = NNTrainer(config['chainer'], **train_config)
 
-            res['valid'] = report['valid']['metrics']
-
-            print(json.dumps(report, ensure_ascii=False))
+    if to_train:
+        trainer.train(iterator)
 
-        if train_config['test_best']:
-            report = {
-                'test': _test_model(model, metrics_functions, iterator,
-                                    train_config.get('batch_size', -1), 'test',
-                                    show_examples=train_config['show_examples'])
-            }
+    res = {}
 
-            res['test'] = report['test']['metrics']
+    if iterator is not None:
+        if to_validate is not None:
+            if evaluation_targets is None:
+                log.warning('"to_validate" parameter is deprecated and will be removed in future versions.'
+                            ' Please, use "evaluation_targets" list instead')
+                evaluation_targets = ['test']
+            else:
+                log.warn('Both "evaluation_targets" and "to_validate" parameters are specified.'
+                         ' "to_validate" is deprecated and will be ignored')
 
-            print(json.dumps(report, ensure_ascii=False))
+        res = trainer.evaluate(iterator, evaluation_targets, print_reports=True)
+        trainer.get_chainer().destroy()
```

Comment:
How about creating trainer's own method `destroy()`? For now train.py contains no `Chainer` abstraction.

Reply:
It can be a little unintuitive.
```python3
my_model = trainer.get_chainer()
trainer.destroy()
result = my_model(batch)
```
This looks like something that should work, but `my_model` is already destroyed